Null giver type, null recipient type

Null giver type


Actual user abc
Request Method POST
User Agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_5) AppleWebKit/537.76.4 (KHTML, like Gecko) Version/6.1.4 Safari/537.76.4
Request Url https://teammatesv4.appspot.com/page/instructorFeedbackQuestionvisibilityMessage
Request Path /page
Request Parameters {showresponsesto::RECEIVER,OWN_TEAM_MEMBERS,RECEIVER_TEAM_MEMBERS,INSTRUCTORS, constSumPointsPerOption::false, constSumPoints::100, showrecipientto::RECEIVER,INSTRUCTORS, rubricNumCols::4, rubricNumRows::2, questiontext::, numofrecipientstype::max, rubricChoice-3::Strongly Disagree, mcqOption-1::, rubricDesc-0-3::Rarely or never responds., constSumOption-1::, constSumToRecipients::false, isNotSureAllowedCheck::on, rubricChoice-0::Strongly Agree, showgiverto::INSTRUCTORS, rubricDesc-0-2::Occasionally responds, but never initiates discussions., mcqOption-0::, constSumOption-0::, rubricDesc-0-1::Takes part in discussions and sometimes initiates discussions., rubricChoice-2::Disagree, rubricChoice-1::Agree, rubricDesc-0-0::Initiates discussions frequently, and engages the team., generatedOptions::NONE//NONE//NONE, questionnum::2//2, rubricSubQn-0::This student participates well in online discussions., courseid::bmnl101, rubricSubQn-1::This student completes assigned tasks on time., noofchoicecreated::2//2//2, rubricDesc-1-3::Rarely or never completes tasks., numofrecipients::1, numscalestep::1, rubricDesc-1-0::Tasks are always completed before the deadline., rubricDesc-1-1::Occasionally misses deadlines., rubricDesc-1-2::Often misses deadlines., numscalemin::1, fsname::session 1, msqOption-0::, receiverLeaderCheckbox::RECEIVER, msqOption-1::, questiontype::CONTRIB, numscalemax::5, user::abc}
Stack Trace



java.lang.AssertionError: Null giver type
	at teammates.common.util.Assumption.fail(Assumption.java:59)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:269)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.extractFeedbackQuestionData(InstructorFeedbackQuestionVisibilityMessageAction.java:39)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.execute(InstructorFeedbackQuestionVisibilityMessageAction.java:18)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:147)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:457)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:230)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

Actual user abc
Request Method POST
User Agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_7_5) AppleWebKit/537.76.4 (KHTML, like Gecko) Version/6.1.4 Safari/537.76.4
Request Url https://teammatesv4.appspot.com/page/instructorFeedbackQuestionvisibilityMessage
Request Path /page
Request Parameters {showresponsesto::RECEIVER,OWN_TEAM_MEMBERS,RECEIVER_TEAM_MEMBERS,INSTRUCTORS, constSumPointsPerOption::false, constSumPoints::100, showrecipientto::RECEIVER,INSTRUCTORS, rubricNumCols::4, rubricNumRows::2, questiontext::, numofrecipientstype::max, rubricChoice-3::Strongly Disagree, mcqOption-1::, rubricDesc-0-3::Rarely or never responds., constSumOption-1::, constSumToRecipients::false, isNotSureAllowedCheck::on, rubricChoice-0::Strongly Agree, showgiverto::INSTRUCTORS, rubricDesc-0-2::Occasionally responds, but never initiates discussions., mcqOption-0::, constSumOption-0::, rubricDesc-0-1::Takes part in discussions and sometimes initiates discussions., rubricChoice-2::Disagree, rubricChoice-1::Agree, rubricDesc-0-0::Initiates discussions frequently, and engages the team., generatedOptions::NONE//NONE//NONE, questionnum::6//6, rubricSubQn-0::This student participates well in online discussions., courseid::bmnl101, rubricSubQn-1::This student completes assigned tasks on time., noofchoicecreated::2//2//2, rubricDesc-1-3::Rarely or never completes tasks., numofrecipients::1, numscalestep::1, rubricDesc-1-0::Tasks are always completed before the deadline., rubricDesc-1-1::Occasionally misses deadlines., rubricDesc-1-2::Often misses deadlines., numscalemin::1, fsname::session 1, msqOption-0::, receiverLeaderCheckbox::RECEIVER, msqOption-1::, questiontype::CONTRIB, numscalemax::5, givertype::STUDENTS, user::abc}
Stack Trace



java.lang.AssertionError: Null recipient type
	at teammates.common.util.Assumption.fail(Assumption.java:59)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:269)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.extractFeedbackQuestionData(InstructorFeedbackQuestionVisibilityMessageAction.java:47)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.execute(InstructorFeedbackQuestionVisibilityMessageAction.java:18)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:147)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:457)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:230)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

Null recipient type


Actual user xyz
Request Method POST
User Agent Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/601.2.7 (KHTML, like Gecko) Version/9.0.1 Safari/601.2.7
Request Url https://teammatesv4.appspot.com/page/instructorFeedbackQuestionvisibilityMessage
Request Path /page
Request Parameters {showresponsesto::RECEIVER,OWN_TEAM_MEMBERS,RECEIVER_TEAM_MEMBERS,INSTRUCTORS, constSumPointsPerOption::false, constSumPoints::100, showrecipientto::RECEIVER,INSTRUCTORS, rubricNumCols::4, rubricNumRows::2, questiontext::, numofrecipientstype::max, rubricChoice-3::Strongly Disagree, mcqOption-1::, rubricDesc-0-3::Rarely or never responds., constSumOption-1::, constSumToRecipients::false, isNotSureAllowedCheck::on, rubricChoice-0::Strongly Agree, showgiverto::INSTRUCTORS, rubricDesc-0-2::Occasionally responds, but never initiates discussions., mcqOption-0::, constSumOption-0::, rubricDesc-0-1::Takes part in discussions and sometimes initiates discussions., rubricChoice-2::Disagree, rubricChoice-1::Agree, rubricDesc-0-0::Initiates discussions frequently, and engages the team., generatedOptions::NONE//NONE//NONE, questionnum::5//5, rubricSubQn-0::This student participates well in online discussions., courseid::GF25A-01, rubricSubQn-1::This student completes assigned tasks on time., noofchoicecreated::2//2//2, rubricDesc-1-3::Rarely or never completes tasks., numofrecipients::1, numscalestep::1, rubricDesc-1-0::Tasks are always completed before the deadline., rubricDesc-1-1::Occasionally misses deadlines., rubricDesc-1-2::Often misses deadlines., numscalemin::1, fsname::Peer evaluation for the Wilford case, msqOption-0::, receiverLeaderCheckbox::RECEIVER, msqOption-1::, questiontype::CONTRIB, numscalemax::5, givertype::STUDENTS, user::xyz}
Stack Trace



java.lang.AssertionError: Null recipient type
	at teammates.common.util.Assumption.fail(Assumption.java:59)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:269)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.extractFeedbackQuestionData(InstructorFeedbackQuestionVisibilityMessageAction.java:47)
	at teammates.ui.controller.InstructorFeedbackQuestionVisibilityMessageAction.execute(InstructorFeedbackQuestionVisibilityMessageAction.java:18)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:337)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:59)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:637)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:37)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:260)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:78)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:147)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:457)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:230)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:235)
	at java.lang.Thread.run(Thread.java:745)

|||/page/instructorFeedbackQuestionvisibilityMessage|||Unknown20151026165410150

--- a/src/main/webapp/js/instructorFeedbackEdit.js
+++ b/src/main/webapp/js/instructorFeedbackEdit.js
@@ -1313,8 +1313,8 @@ function fixContribQnGiverRecipient(questionNumber) {
     $giverType.find('option').not('[value="STUDENTS"]').prop('disabled', true);
     $recipientType.find('option').not('[value="OWN_TEAM_MEMBERS_INCLUDING_SELF"]').prop('disabled', true);
 
-    $giverType.find('option').filter('[value="STUDENTS"]').attr('selected','selected');
-    $recipientType.find('option').filter('[value="OWN_TEAM_MEMBERS_INCLUDING_SELF"]').attr('selected','selected');
+    $giverType.find('option').filter('[value="STUDENTS"]').prop('selected', true);
+    $recipientType.find('option').filter('[value="OWN_TEAM_MEMBERS_INCLUDING_SELF"]').prop('selected', true);
 }
 
 /**

